doc/stdenv/stdenv.chapter.md

```diff
@@ -309,13 +309,51 @@ The attribute can also contain a list, a script followed by arguments to be pass
+Finally, the attribute can be an attribute set, which, in addition to the command, can optionally provide the canonical attribute path to use for the list of extra features the script supports.
+
+```nix
+passthru.updateScript = {
+  command = [ ../../update.sh pname ];
+  attrPath = pname;
+  supportedFeatures = [ … ];
+};
+```
+
+##### How update scripts are executed? {#var-passthru-updateScript-execution}
+
+For information about how to run update scripts, run `nix-shell maintainers/scripts/update.nix` in the root of Nixpkgs repository. The script offers several modes for finding and executing update scripts.
+
+Each script will be run with the `UPDATE_NIX_NAME`, `UPDATE_NIX_PNAME`, `UPDATE_NIX_OLD_VERSION` and `UPDATE_NIX_ATTR_PATH` environment variables set respectively to the name, pname, old version and attribute path of the package it is supposed to update.
 
 ::: {.note}
 The script will be usually run from the root of the Nixpkgs repository but you should not rely on that. Also note that the update scripts will be run in parallel by default; you should avoid running `git commit` or any other commands that cannot handle that.
 :::
 
-For information about how to run the updates, execute `nix-shell maintainers/scripts/update.nix`.
+`maintainers/scripts/update.nix` also supports automatically creating commits when running it with `--argstr commit true`. Update scripts can opt to override the commit contents by adding `commit` to `supportedFeatures` and [modifying the script accordingly](#var-passthru-updateScript-commit). It might be useful, for example, for including news link in the commit message.
+
+##### Supported features {#var-passthru-updateScript-supported-features}
+###### `commit` {#var-passthru-updateScript-commit}
+
+When support of this feature is declared, whenever the update script exits with `0` return status, it is expected to print a JSON list containing an object for each updated attribute. When `update.nix` is run in commit mode, each of the objects will result in a separate commit being created. Empty list can be returned when the script did not update any files, for example, when the package is already at the latest version.
+
+Each object must contain the keys shown below:
+
+```json
+[
+  {
+    "attrPath": "volume_key",
+    "oldVersion": "0.3.11",
+    "newVersion": "0.3.12",
+    "files": [
+      "/path/to/nixpkgs/pkgs/development/libraries/volume-key/default.nix"
+    ]
+  }
+]
+```
+
+Unless the returned array contains exactly one object (e.g. `[{}]`), in which case all keys can be omitted and will be determined automatically.
+
+In addition to the keys listed above `commitBody` key can be provided, which will append extra content to the default commit message (useful for adding changelog links). Or `commitMessage` key can override the commit message completely.
```

Comment:
As above, this should be a list of keys, each with a description, followed by a full example of how they are used. It's much harder to read prose than skim a list, and no one really reads things when looking up information.

Reply:
Done.